pkg/buildscript/buildscript.go

```diff
@@ -0,0 +1,179 @@
+package buildscript
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/pkg/buildscript/internal/buildexpression"
+	"github.com/ActiveState/cli/pkg/buildscript/internal/raw"
+	"github.com/ActiveState/cli/pkg/platform/api/buildplanner/types"
+	"github.com/ActiveState/cli/pkg/platform/api/mono/mono_models"
+	"github.com/go-openapi/strfmt"
+)
+
+// BuildScript is what we want consuming code to work with. This specifically makes the raw presentation private as no
+// consuming code should ever be looking at the raw representation, instead this package should facilitate the use-case
+// of the consuming code through convenience methods that are easy to understand and work with.
+type BuildScript struct {
+	// buildexpression is what we do all our modifications on. We will be planning work to move this to the raw type
+	// instead, but for now this is where most of the actual low level modification logic is done.
+	buildexpression *buildexpression.BuildExpression
+	atTime          *time.Time
+}
+
+func New() (*BuildScript, error) {
+	expr, err := buildexpression.New()
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not create empty build expression")
+	}
+	script, err := unmarshalBuildExpressionTyped(expr, nil)
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not create empty build expression")
+	}
+	return script, nil
+}
+
+// Unmarshal will parse a buildscript from its presentation on disk
+// This needs to unmarshal the ascript representation, and then convert that representation into a build expression
+func Unmarshal(data []byte) (*BuildScript, error) {
+	raw, err := raw.Unmarshal(data)
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not unmarshal buildscript")
+	}
+
+	be, err := raw.MarshalBuildExpression()
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not marshal build expression from raw")
+	}
+
+	expr, err := buildexpression.Unmarshal(be)
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not unmarshal build expression")
+	}
+
+	return &BuildScript{expr, raw.AtTime}, nil
+}
+
+// UnmarshalBuildExpression will create buildscript using an existing build expression
+func UnmarshalBuildExpression(b []byte, atTime *time.Time) (*BuildScript, error) {
```

Comment:
Please add a comment about the atTime parameter and why it's needed or optional for this function.